src/worker/queue.ts

```diff
@@ -48,3 +59,83 @@ export function startQueue(
 
     return worker
 }
+
+function startQueueKafka(
+    server: PluginsServer,
+    processEvent: (event: PluginEvent) => Promise<PluginEvent | null>
+): Queue {
+    const eventsProcessor = new EventsProcessor(server)
+
+    eventsProcessor.kafkaConsumer.on('ready', () => {
+        eventsProcessor.kafkaConsumer.subscribe([KAFKA_EVENTS_WAL])
+        // consume event messages in batches of 1000 every 50 ms
```

Comment:
This scares me. 

I would still greatly prefer a solution like https://github.com/PostHog/posthog-plugin-server/issues/62 - where we pause the consumption when the piscina queue gets full and resume once we have capacity again. Like it is now, if the server gets any backlog, it'll just consume messages endlessly. They will all be `await`ing... and we will probably run out of memory and explode.

Reply:
With an interval we can just `return` immediately if some `server.isQueueFull` (or something like that) is true, so that should be simple enough.

Reply:
Indeed, that's a simple and clever workaround that will block the explosion. The next step will then still be to use `process.nextTick` and immediately get more events once the queue frees up.